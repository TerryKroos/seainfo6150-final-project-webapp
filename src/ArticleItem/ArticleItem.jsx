import React from 'react';
import styles from "./ArticleItem.module.css";
import {BrowserRouter as Router, Link} from 'react-router-dom';
import { useHistory } from "react-router-dom";


const ArticleItem = (props)=>{

    const history = useHistory();
    const handleSubmit = () => {
        history.push('/article', {params: props.article})
    }
    return(
        <div className={styles.container} onClick={() =>
           {handleSubmit()}}>
           <img src={props.article.image} alt="article img"/>
           <div className={styles.articletext}>
              <time datetime="2018-11-22">{props.article.date}</time>
              <Link className={styles.link}>
              {props.article.heading}</Link>
           </div>
        </div>
        );
}

export default ArticleItem;