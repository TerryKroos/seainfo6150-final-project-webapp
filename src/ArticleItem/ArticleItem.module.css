.container {
    height: 70%;
    width: 100%;
    background-color: #F0F0F0;
    cursor: pointer;
    color: #556575;
}
img {
 width: 100%;   
 height: 100%;
 max-height: 300px;

}
.articletext {
    display: flex;
    flex-direction: column;
    align-items: center;
    background-color: #F0F0F0;
    height: 50%;
    width:100%;
    font-size: 14px;
}
.link{
    margin-top: 10px;
    text-decoration: none;
    color: #556575;
    cursor: pointer;
  }
  .link:hover{
    color: #282828; 
  }

   

