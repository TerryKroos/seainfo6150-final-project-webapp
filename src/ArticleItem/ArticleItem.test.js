
import React from "react";
import ArticleItem from "./ArticleItem";
import { render } from "@testing-library/react";

describe("ArticleItem tests", () => {
    it("renders correctly", () => {
      const { container } = render(<ArticleItem article={{}}/>);
      expect(container).toMatchSnapshot();
    });
  
  });