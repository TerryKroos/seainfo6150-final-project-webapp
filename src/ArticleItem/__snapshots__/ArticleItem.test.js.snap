// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleItem tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <img
      alt="article img"
    />
    <div
      class="articletext"
    >
      <time
        datetime="2018-11-22"
      />
      <a
        class="link"
        href=""
      />
    </div>
  </div>
</div>
`;
